### PR TITLE
Add precommit hooks for M1 mac

### LINKED ISSUE #25: Add support for building M1 wheel files in Github Actions
Hey, I noticed in [workflows file](https://github.com/deepmind/lab2d/tree/main/.github/workflows) that it builds the wheels for various x86_64 platforms. 

GitHub has recently [announced](https://github.blog/changelog/2022-08-09-github-actions-self-hosted-runners-now-support-apple-m1-hardware/) support for M1 macs in actions as well, so would it be possible to build M1 wheel files as well? 

This will help in avoiding the hassle of building it from source.

### PR DESCRIPTION
**Pull request by ashok-arora**
Resolves #25.

### CODE REVIEW

File: .github/workflows/precommit.yml [hunk 1]
```diff
@@ -30,6 +30,7 @@ jobs:
           - { name: 'Linux LLVM+libstdc++', os: 'ubuntu-22.04', cc: clang, cxx: clang++, config: ''}
           - { name: 'Linux LLVM+libc++', os: 'ubuntu-22.04', cc: clang, cxx: clang++, config: --config=libc++ }
           - { name: 'MacOS LLVM+libc++', os: 'macos-12', cc: clang, cxx: clang++, config: --config=libc++ }
+          - { name: 'M1 MacOS LLVM+libc++', os: 'macos-latest', cc: clang, cxx: clang++, config: --config=libc++ }
```
Comment (tkoeppe) [inline .github/workflows/precommit.yml:33]: Change these two to `MacOS-x86_64` and `MacOS-ARM` maybe?

Comment (tkoeppe) [inline .github/workflows/precommit.yml:33]: Wait, `macos-lastest` is the only way to request an ARM machine? That seems awfully brittle. What if latest starts meaning something else? And how do we retain the x86_64 versions?

Could you provide some refefences for how the actual hardware is selected for the action runners? Then let's try to find a robust and stable way to get that configuration.

Comment (ashok-arora) [reply .github/workflows/precommit.yml:33]: > Wait, `macos-lastest` is the only way to request an ARM machine? That seems awfully brittle. What if latest starts meaning something else? And how do we retain the x86_64 versions?
> 
> 
> 
> Could you provide some refefences for how the actual hardware is selected for the action runners? Then let's try to find a robust and stable way to get that configuration.

Here's the doc that motivated me to work on it: https://github.com/github/roadmap/issues/528

Comment (ashok-arora) [reply .github/workflows/precommit.yml:33]: I haven't tested them out personally yet, soon I'll create a dummy repository to test out all the various options with focus on specific version instead of the `-latest` tag to avoid the brittle nature.

Comment (tkoeppe) [reply .github/workflows/precommit.yml:33]: You can just use this pull request to run the actions if you like...

Comment (ashok-arora) [reply .github/workflows/precommit.yml:33]: Hi @tkoeppe, so I tried the GitHub actions thing and there's a catch. The Github Actions hosted runner is in **public beta** so not everyone got access to it I assume. A completely public option is to use a self-hosted runner deployed on our own M1 machine, do we have any available M1 macs that could do the job?

Also, as an alternative, I can volunteer to build the project on my mac and prepare the `pip` file for it which can be distributed easily. Would that be okay?

File: third_party/rl_api/BUILD
Comment (tkoeppe) [inline third_party/rl_api/BUILD:1]: Revert the changes in this file. With the LD64 linker, the status quo should work fine.

**Comment by tkoeppe** [comment]
So, I added `macos-13` experimentally to the wheel maker action, and have a look:

https://github.com/deepmind/lab2d/releases/tag/untagged-14c0e5455b045948a58d

The artifact is dmlab2d-1.0-cp39-cp39-macosx_13_0_x86_64.whl, and is an x86_64 binary:

```
file dmlab2d_pybind.so
dmlab2d_pybind.so: Mach-O 64-bit x86_64 dynamically linked shared library, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|WEAK_DEFINES|BINDS_TO_WEAK|NO_REEXPORTED_DYLIBS|HAS_TLV_DESCRIPTORS>
```

So I have no indication that the GitHub runner `macos-13` actually uses ARM hardware.

Could you please clarify?

**Comment by tkoeppe** [comment]
I'm closing this PR for now, since there's no indication that it actually builds ARM code. Please reopen if you want to continue the discussion.

**Comment by ashok-arora** [comment]
Hey @tkoeppe, please see what I had discussed earlier: https://github.com/deepmind/lab2d/pull/26#discussion_r1185463787

Also, the link: https://github.com/deepmind/lab2d/releases/tag/untagged-14c0e5455b045948a58d returns 404.

**Comment by tkoeppe** [comment]
Oh, sorry, because that's not published release. Here, I did make a published one which also shows only x86_64 builds: https://github.com/deepmind/lab2d/releases/tag/release_candidate_2023-05-22

Re the runners: I realize that `macos-13` itself as an environment spec is beta, but what I'm missing is any indication that there will be M1 hardware. The current beta runners (which are available just fine, it's just that their spec isn't stable yet) use x86 hardware. So what I'm missing in this proposal is a concrete mechanism for getting an M1 environment.
